packages/lisk-transaction-pool/src/job.ts

```diff
@@ -0,0 +1,50 @@
+export class Job<T> {
+	private _id: NodeJS.Timer | undefined;
+	private readonly _interval: number;
+	private readonly _job: () => Promise<T>;
+	private _running = false;
+
+	public constructor(
+		context: object,
+		job: () => Promise<T>,
+		interval: number,
+	) {
+		this._interval = interval;
+		this._job = job.bind(context);
```

Comment:
should we bind it for them, or it's caller's responsibility?

Reply:
I thought it's better to pass context to constructor function so we don't have to do it for every new job. Do you prefer the other way around?

Reply:
I think it's more clear to do it outside so it's easier to see what the context is when registering the job